discovery/gossiper.go

```diff
@@ -906,18 +906,21 @@ func (d *AuthenticatedGossiper) networkHandler() {
 			policyUpdate.errResp <- nil
 
 		case announcement := <-d.networkMsgs:
+			// We should only broadcast this message forward if it
+			// wasn't received as part of our initial historical
+			// sync.
+			shouldBroadcast := d.syncMgr.IsGraphSynced()
+
 			switch announcement.msg.(type) {
 			// Channel announcement signatures are amongst the only
 			// messages that we'll process serially.
 			case *lnwire.AnnounceSignatures:
 				emittedAnnouncements := d.processNetworkAnnouncement(
 					announcement,
 				)
-				if emittedAnnouncements != nil {
-					announcements.AddMsgs(
-						emittedAnnouncements...,
-					)
-				}
+				announcements.AddMsgs(
```

Comment:
Why omit the check? It lets us avoid grabbing a mutex FWIW.

Reply:
Good point, fixed.